sky/packages/sky/lib/src/fn3/framework.dart

```diff
@@ -180,11 +210,15 @@ typedef void ElementVisitor(Element element);
 
 const Object _uniqueChild = const Object();
 
+abstract class BuildContext {
+  InheritedWidget inheritedWidgetOfType(Type targetType);
+}
+
 /// Elements are the instantiations of Widget configurations.
 ///
 /// Elements can, in principle, have children. Only subclasses of
 /// RenderObjectElement are allowed to have more than one child.
-abstract class Element<T extends Widget> {
+abstract class Element<T extends Widget> extends BuildContext {
```

Comment:
I'd use "implements", not "extends", since it's not really an "isa". According to Ivan it makes no difference to perf, FWIW.